src/Features/Core/Portable/Diagnostics/Analyzers/RemoveUnnecessaryCastDiagnosticAnalyzerBase.cs

```diff
@@ -3,76 +3,70 @@
 using System.Collections.Immutable;
 using System.Linq;
 using System.Threading;
+using Microsoft.CodeAnalysis.CodeStyle;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.Diagnostics.RemoveUnnecessaryCast
 {
-    internal abstract class RemoveUnnecessaryCastDiagnosticAnalyzerBase<TLanguageKindEnum> : DiagnosticAnalyzer, IBuiltInAnalyzer where TLanguageKindEnum : struct
+    internal abstract class RemoveUnnecessaryCastDiagnosticAnalyzerBase<
+        TLanguageKindEnum,
+        TCastExpression> : AbstractCodeStyleDiagnosticAnalyzer
```

Comment:
AbstractCodeStyleDiagnosticAnalyzer gives us most of the scaffolding we need.  Allowing us to remove huge amounts of repetitive and unnecessary code.